convokit/coordination/coordination.py

```diff
@@ -111,7 +111,11 @@ def transform(self, corpus: Corpus) -> Corpus:
         for (speaker, target), score in pair_scores.items():
             if self.coordination_attribute_name not in speaker.meta:
                 speaker.meta[self.coordination_attribute_name] = {}
-            speaker.meta[self.coordination_attribute_name][target.id] = score
+            # Avoid mutability of dict in metadata fields
+            temp_dict = {}
+            for k, v in speaker.meta[self.coordination_attribute_name]:
+                temp_dict[k] = v
+            speaker.meta[self.coordination_attribute_name] = temp_dict
```

Comment:
- Since this is in the for-loop, this will replace the dict a bunch of times
- Comment should specify that we're avoiding mutability for the sake of DB corpus

That aside, wouldn't it be better design to get `ConvoKitMeta` to carry out a proper update here regardless of whether in memory/DB mode? Disallowing basic Python patterns like updating a sub-dictionary seems like too much of a gotcha.

Reply:
> * Since this is in the for-loop, this will replace the dict a bunch of times
> * Comment should specify that we're avoiding mutability for the sake of DB corpus

Thank you for this careful review Caleb! I have modified the code to make the implementation more efficient (as discussed with Jonathan and Prof. DNM, Thanks!) : By saving all score update to (speaker, target) pairs together in a dict, update the dict if such (speaker, target) pair has score updated again, and only replacing the dict by the end when all changes are final.

Reply:
> That aside, wouldn't it be better design to get `ConvoKitMeta` to carry out a proper update here regardless of whether in memory/DB mode? Disallowing basic Python patterns like updating a sub-dictionary seems like too much of a gotcha.

From my understanding, the issue is that when updating a sub-dictionary in metadata fields, there is no way for system to know there is a change to the metadata, so it would not trigger an update to DB, causing data loss. That's why we here would like to eliminate mutability of mutable types for metadata fields. I'm not sure if I understand your suggestion correctly, if you have any better design idea, Please let me know! Thanks a lot!